docs/gcp_workload_identity.md

```diff
@@ -0,0 +1,122 @@
+   ```
+4. Create an install-config.yaml
+   ```
+   ./openshift-install create install-config
+   ```
+5. Make sure that we install the cluster in Manual mode
+   ```
+   echo "credentialsMode: Manual" >> install-config.yaml
+   ``` 
+6. Create install manifests
+   ```
+   ./openshift-install create manifests   
+   ```
+7. Create GCP resources using [ccoctl](./ccoctl.md) tool (you will need GCP credentials with sufficient permissions). Below command will generate public/private ServiceAccount signing keys, create the cloud storage bucket, upload the OIDC config into the bucket, set up a workload identity pool/provider and create IAM service account for each GCP Credentials Request. It will also dump the files needed by the installer in the `output_dir` directory
+   ```
+   ccoctl gcp create-all --name=<gcp_infra_name> --region=<gcp_region> --project=<gcp-project-id> --credentials-requests-dir=/path/to/credreqs/directory/created/in/step/2 --output-dir=<output_dir>
+   ```
+8. Copy the manifests created in the step 7 and put them in the same location as install-config.yaml in the `manifests` directory
+   ```
+   cp _output/manifests/* /path/to/dir/with/install-config.yaml/manifests/
+   ```
+9. Copy the private key for the ServiceAccount signer and put it in the same location as install-config.yaml
+   ```
+   cp -a _output/tls /path/to/dir/with/install-config.yaml
+   ```
+10. Run the OpenShift installer
+    ```
+    ./openshift-install create cluster --log-level=debug
+    ```
+
+### Post install verification
+
+1. Connect to the newly installed cluster and verify that the OpenShift cluster does not have `root` credentials. Below command should throw secret not found error
+   ```yaml
+   oc get secrets -n kube-system gcp-credentials
+   ```
+2. Verify that components are assuming the IAM Service Account specified in the secret manifests, instead of creds minted by the cloud-credential-operator. Below command should show you the `type` as `external_account` and `service_account_impersonation_url` should contain the email of the service account used by image registry operator.
+   ```yaml
+   oc get secrets -n openshift-image-registry installer-cloud-credentials -o json | jq -r '.data."service_account.json"' | base64 -d
+   ```
+   sample output of the above command
+   ```json
+   {
+      "type": "external_account",
+      "audience": "//iam.googleapis.com/projects/123456789/locations/global/workloadIdentityPools/test-pool/providers/test-provider",
+      "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+      "token_url": "https://sts.googleapis.com/v1/token",
+      "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/test-openshift-image-registry-42ssv@test-project.iam.gserviceaccount.com:generateAccessToken",
+      "credential_source": {
+         "file": "var/run/secrets/openshift/serviceaccount/token",
```

Comment:
whoops, you took my literal suggestion which had a typo :)
```suggestion
         "file": "/var/run/secrets/openshift/serviceaccount/token",
```

Reply:
fixed